include/emitc/emitc_tosa.h

```diff
@@ -178,6 +178,91 @@ Dest conv2d(Src input, Weights weights, Tensor1D<int64_t, 4> padding,
   return output;
 }
 
+// DepthwiseConv2DOp
+template <typename Dest, typename Src, typename Weights>
+Dest depthwise_conv2d(Src input, Weights weights, Tensor1D<int64_t, 4> padding,
+                      Tensor1D<int64_t, 2> stride,
+                      Tensor1D<int64_t, 2> dilation) {
+  // Input is [N,IH,IW,IC], weights
+  // are [KH,KW,CIN,M] and output is [N,H,W,C*M].
```

Comment:
Optional: You could add static_asserts that the dimensions match
```c
Src::dim(0) == Dest::dim(0);
Src::dim(3) == Weights::dim(2);
...
```

Reply:
That's to obvious :P